### PR TITLE
Todo list tests: part 2

### PR DESCRIPTION
**Pull request by kimothokamau**
This pull request contains tests for: 
- drag and drop
- editing tasks
- clearing all completed tasks
- marking tasks as completed

### CODE REVIEW

File: __tests__/edit.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      description: 'Sample 2',
+      completed: false,
+      index: 1,
+    },
+    {
+      description: 'Sample 3',
+      completed: false,
+      index: 2,
+    },
+  ];
```
Comment (dasileker) [inline __tests__/edit.test.js:26]: - [x] Please Add a test that really tests the function not an array of objects, and I do see that you are repeating the same thing for the `completed.test.js` + `clearAll.test.js`.

File: __tests__/completed.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  if (element.dispatchEvent(change)) {
+    list[0].completed = true;
+    mockStorage.setList(list);
+    mockDOM.displayTask(mockStorage.list);
+  }
+
+  test('Task completion is updated successfully', () => {
+    expect(list[0].completed).toBeTruthy();
+  });
```
Comment (dasileker) [inline __tests__/completed.test.js:45]: - Please add negative tests also.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hello @team** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
### **Check the comments under the review 👇.**

## Optional suggestions
- Your tests cases should have **`tests for most pure functions.`**

You can also consider:
- You can use the jest library to see how to make your test more efficient and more elegant go here 👉 [jest tests library](https://jestjs.io/docs/asynchronous)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @eri8-9 @kimothokamau @francisuloko  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kimothokamau** [comment]
Thanks so much @mricanho 
Happy coding as well

**Pull request merged**